samples/CMakeLists.txt

```diff
@@ -5,6 +5,20 @@
 #
 
 if (CONFIG_SPM)
+  get_cmake_property(VARIABLES              VARIABLES)
+  get_cmake_property(VARIABLES_CACHED CACHE_VARIABLES)
+
+  set(regex "^CONFIG_SPM_SERVICE.+")
+
+  list(FILTER VARIABLES        INCLUDE REGEX ${regex})
+  list(FILTER VARIABLES_CACHED INCLUDE REGEX ${regex})
+
+  foreach(var_name ${VARIABLES} ${VARIABLES_CACHED})
+    set(spm_${var_name} ${${var_name}})
```

Comment:
So, if user did: `-Dspm_CONFIG_SPM_SERVICE<something>=<value>` that will be lost and overwritten with the value of `CONFIG_SPM_SERVICE<something>`.

Sounds like a flaw.

Reply:
In the case of these configs, any mismatch in configuration between SPM and app is an error, so in this case, it's "desired". Ideally, there should be a way to prevent setting the config in SPM, but I opted to just mention it in the help text.

Reply:
accepted.
Probably we should warn the user then, in case he has already defined `spm_CONFIG_SPM_SERVICE<something>` before overwriting the value.

But that  can be done as enhancement.